src/main/java/com/cookingchef/dao/CartDAO.java

```diff
@@ -10,38 +10,38 @@ public interface CartDAO {
     /**
      * Add element to a cart in the database
      *
-     * @param cart The cart object that you want to register.
+     * @param cartEntry The cart object that you want to register.
      * @return The id of the newly created cart
      */
-    public Optional<Integer> addElementIntoCartInDb(Cart cart) throws SQLException;
+    public Optional<Integer> addElementIntoCartInDb(CartEntry cartEntry) throws SQLException;
 
     /**
      * Update the quantity of element of the cart in the database.
      *
-     * @param cart The cart object that you want to update in the database.
+     * @param cartEntry The cart object that you want to update in the database.
      */
-    public void updateCartInDb(Cart cart) throws SQLException;
+    public void updateCartInDb(CartEntry cartEntry) throws SQLException;
 
     /**
      * Removes an element of the cart from the database.
      *
-     * @param cart The cart object to be removed from the database.
+     * @param cartEntry The cart object to be removed from the database.
      */
-    public void removeCartFromDb(Cart cart) throws SQLException;
+    public void removeCartFromDb(CartEntry cartEntry) throws SQLException;
 
     /**
      * Get a cart by id
      *
      * @param id The id of the cart you want to get.
      * @return The cart if it exists, otherwise, returns an empty option
      */
-    public Optional<Cart> getCartById(int id) throws SQLException;
+    public Optional<CartEntry> getCartById(int id) throws SQLException;
```

Comment:
You should add the `ingredient_id` to fully identify the CartEntry
```suggestion
    public Optional<CartEntry> getCartById(int userId, int ingredientId) throws SQLException;
```